### PR TITLE
docs: regular update (early August)

### LINKED ISSUE #487: cmd ref: advanced examples for import (and get)
> The previous PR was too long. We're extracting this part, which is pending and hopefully the last related to `dvc import` and `dvc get`. See https://github.com/iterative/dvc.org/pull/464#pullrequestreview-259871929

- [x] Note that the `import-url` ref and `external-dependencies` guide also needs an update (See https://github.com/iterative/dvc.org/pull/464#pullrequestreview-257721375).

- [x] Create more sample repos for advanced `import` and possibly `get` command example/s. See https://github.com/iterative/dvc.org/issues/487#issuecomment-517520838 below.

- [x] Actually create/update advanced examples for import/get commands.

- ~~Check the `example-pipeline` carefully. It looks like it might be broken now. (Check both examples in fact.)~~ Moved to #564

- [x] Lets also update get started, work on use cases, etc. to incorporate these important new commands (include also `dvc update`).

### PR DESCRIPTION
**Pull request by jorgeorpinel**

### CODE REVIEW

File: static/docs/commands-reference/import-url.md [hunk 1]
```diff
@@ -284,6 +284,7 @@ $ tree
     ├── featurization.py
     ├── prepare.py
     └── train.py
+```
```
Comment (jorgeorpinel) [inline static/docs/commands-reference/import-url.md]: This whole commit is a typo... Sry

Fixed in https://github.com/iterative/dvc.org/pull/522/commits/be18aad12fa66a917d45a8200fa72150b27a10d0#diff-0a462fb4a411db0aa4af5e0a1036b7f0L291-L293

File: src/Documentation/glossary.js [hunk 2]
```diff
@@ -5,15 +5,15 @@ export default {
     'relevant DVC concepts and terminologies which are frequently used.',
   contents: [
     {
-      name: 'Workspace directory',
+      name: 'DVC Workspace',
```
Comment (shcheklein) [inline src/Documentation/glossary.js]: let's keep the previous one. Actually, we should change the description below. I think we should be emphasizing DVC here. For example, I can have workspace (Git repo, or even without Git) with my code and data without any DVC. So, workspace is broader, more generic term. It contains code, data, models, DVC-files if DVC is enabled.

File: static/docs/commands-reference/checkout.md [hunk 3]
```diff
@@ -103,36 +103,20 @@ be pulled from a remote cache using `dvc pull`.
 
 ## Examples
 
-To explore `dvc checkout` let's consider a simple workspace with several
-[stages](/doc/commands-reference/run), and a few Git tags. Then with
-`git checkout` and `dvc checkout` we can see what happens as we shift from tag
-to tag.
+Let's employ a simple workspace with some stages and a few Git tags, such as our
```
Comment (shcheklein) [inline static/docs/commands-reference/checkout.md]: `some code, data, ML models, and pipelines` is better than `some stages`

File: static/docs/commands-reference/commit.md [hunk 4]
```diff
@@ -90,37 +90,32 @@ into play. It handles that last step of adding the file to the DVC cache.
 
 ## Examples
 
-To explore `dvc commit` let's consider a simple pipeline with several stages:
-the example workspace used in the [Get Started](/doc/get-started) section.
+Let's employ a simple workspace with several stages, such as the DVC project
```
Comment (shcheklein) [inline static/docs/commands-reference/commit.md]: same here, see checkout

File: src/Documentation/glossary.js (hunk 2 above)
Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: Isn't calling it "workspace directory" redundant? The workspace IS a directory. We never call it that in the docs. I vote for just "Workspace" but lmk if you prefer "Workspace Directory" and I'll restore it.

Updated description, better? Not sure I'm understanding your comment 100%.

File: static/docs/commands-reference/commit.md (hunk 4 above)
Comment (jorgeorpinel) [reply static/docs/commands-reference/commit.md]: And also in `dvc fetch`.

File: static/docs/commands-reference/checkout.md (hunk 3 above)
Comment (jorgeorpinel) [reply static/docs/commands-reference/checkout.md]: Sure. Updated in f1dc617

File: src/Documentation/glossary.js (hunk 2 above)
Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: Oh... So you mean the "workspace" is a general concept that may or may not include DVC? That would be a new way of understanding the term for me! I thought we only call it "workspace" in the context of an existing DVC project (a directory that has been `dvc init`ialized).

Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: Ping @shcheklein

Comment (shcheklein) [reply src/Documentation/glossary.js]: hey! yep, `directory` is redundant for sure.

Workspace is a more general term than DVC project.

For example, in get started it's pretty reasonable to say:

`Without DVC you have to manually put data in your workspace into .gitignore.`

Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: I see. Updated in 686561b but I may need to review also all the existing uses of the term "workspace" to make sure its meaning is consistent...

p.s. Also made some updates to the get-started including your note above in fc28eb0.

File: src/Documentation/glossary.js [hunk 5]
```diff
@@ -5,15 +5,18 @@ export default {
...
-        'The **workspace** contains all of your DVC **project** files and ' +
-        "directories. It's typically also a Git **repository**. See also " +
-        '[`dvc init`](/doc/commands-reference/init).'
+      desc: `
+By "workspace" we refer to the directory containing all your project files. For
+example raw datasets, source code, ML models, etc. A workspace becomes a DVC
+project when [\`dvc init\`](/doc/commands-reference/init) is run, and
+[DVC-files](/doc/user-guide/dvc-file-format) are created in it. It\s typically
+also a Git repository.
+      `
```
[the remark this answers is not in the document]
Comment (shcheklein) [reply src/Documentation/glossary.js]: looks a bit strange, but let's try 👍 ... does prettier handle this properly?

**Comment by shcheklein** [comment]
@jorgeorpinel please, resolve the conflicts

File: static/docs/get-started/add-files.md [hunk 6]
```diff
@@ -16,47 +21,42 @@ into `data` subdirectory. To download, right-click
...
+DVC stores information about the added data in a special **DVC-file** that has a
+human-readable [format](/doc/user-guide/dvc-file-format). It can be committed to
+Git:
 
 ```dvc
 $ git add data/.gitignore data/data.xml.dvc
 $ git commit -m "add raw data to DVC"
 ```
 
+> Committing these special files to Git allows us to tack different versions of
```
Comment (shcheklein) [inline static/docs/get-started/add-files.md]: I would not make it a note. It's quite important piece of information.

File: src/Documentation/glossary.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: Good question. I don't think `prettier` is being run on js files automatically with the pre-commit hook, is it? It hasn't complained about any js file in my experience.

I manually ran `npm run prettier-js` and `glossary.js` passed.

[the remark this answers is not in the document]
Comment (shcheklein) [reply src/Documentation/glossary.js]: It is running. I would double check if it automatically wraps the lines in both cases properly before making a decision here

**Comment by jorgeorpinel** [comment]
I'm waiting for all the changes to be pushed and approved first, so that ideally I only have to resolve conflicts once at the very end. Note that I'm also switching back to merging from upstream instead of rebase (another reason to leave it for last). All this per as agreed in a recent conversation we had with the team on Slack 🙂

**Comment by shcheklein** [comment]
@jorgeorpinel sounds good! I just thought that we are almost ready to merge this one.

File: src/Documentation/glossary.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: If your question was whether the linter will fix lines that go over 80 chars, no. Neither in the previous or in this case does it fix them with neither the hook nor the `prettier-js` script 🙁. It seems to leave strings alone. Is that another issue to report to Aman?

[the remark this answers is not in the document]
Comment (shcheklein) [reply src/Documentation/glossary.js]: we can def check if it's possible to configure it to fix them. In any case, I don't mind the format you suggested, its' def easier to support manually

[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: cc @algomaster99 what do you think? (about `pretty-quick` being able to warp strings in js files)

[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply src/Documentation/glossary.js]: > looks a bit strange, but let's try

Applied new format to all glossary entries in 8781a71

File: static/docs/get-started/add-files.md (hunk 6 above)
Comment (jorgeorpinel) [reply static/docs/get-started/add-files.md]: You're right, made into regular paragraph in 328b2ec

**Comment by jorgeorpinel** [comment]
> I just thought that we are almost ready to merge this one.

I wish haha... I have lots of changes after reviewing the usage of the term "workspace" per a comment somewhere in this PR

BUT I just realized that should go in it's own PR! So I saved that work separately and just pushed changes for the pending comments above. Please check @shcheklein (:

I'll start resolving conflicts now.

**Pull request merged**